appender.go

```diff
@@ -214,6 +226,9 @@ func (a *Appender) Close() error {
 	}
 
 	// Destroy all appender data and the appender.
+	if errFlush != nil {
+		_ = a.Clear()
```

Comment:
Maybe we should still try to track the error here? I understand why we assume it always succeeds logically here, but if it does not, the users have a kind of "undefined" state and might even want to restart or so?